go-controller/pkg/node/management-port-dpu.go

```diff
@@ -179,20 +175,27 @@ func newManagementPortNetdev(hostSubnets []*net.IPNet) ManagementPort {
 	}
 }
 
-func (mp *managementPortNetdev) Create(nodeAnnotator kube.Annotator, waiter *startupWaiter) (*managementPortConfig, error) {
-	klog.Infof("Lookup netdevice link and existing management port")
-	// get netdev that is used for management port.
-	link, err := util.GetNetLinkOps().LinkByName(mp.netdevName)
+func getManagmentPortNetDev(netdevName string) (netlink.Link, error) {
+	link, err := util.GetNetLinkOps().LinkByName(netdevName)
 	if err != nil {
 		if !util.GetNetLinkOps().IsLinkNotFoundError(err) {
-			return nil, fmt.Errorf("failed to lookup %s link: %v", mp.netdevName, err)
+			return nil, fmt.Errorf("failed to lookup %s link: %v", netdevName, err)
 		}
 		// this may not the first time invoked on the node after reboot
 		// netdev may have already been renamed to ovn-k8s-mp0.
 		link, err = util.GetNetLinkOps().LinkByName(types.K8sMgmtIntfName)
 		if err != nil {
-			return nil, fmt.Errorf("failed to get link device for %s. %v", mp.netdevName, err)
+			return nil, fmt.Errorf("failed to get link device for %s. %v", netdevName, err)
 		}
+	}
+	return link, err
+}
+
+func (mp *managementPortNetdev) Create(nodeAnnotator kube.Annotator, waiter *startupWaiter) (*managementPortConfig, error) {
+	klog.Infof("Lookup netdevice link and existing management port")
+	link, err := getManagmentPortNetDev(mp.netdevName)
+	if err != nil {
+		return nil, err
 	} else if mp.netdevName != types.K8sMgmtIntfName {
```

Comment:
This are incorrect anymore.
`else if` will always succeed because netdev name in the configuration differs from `types.K8sMgmtIntfName`.
Change to `} else if link.Attrs().Name != types.K8sMgmtIntfName {`

Reply:
What happens here in Your case. At first run (newly configured node) - control will go here to find old ovn-k8s-mp0 interface and wouldn't find it. But during 2nd run, lets say on pod restart, it will find it because it is our newly configured management port. It will rename it to generic name (`net0417...`), but because we have a Link to it, mgmt port will be configured correctly

Reply:
As per your request, I've changed the code to `link.Attrs().Name != types.K8sMgmtIntfName`. The same log message still appears (even after multiple restarts):

```
I0419 10:49:40.559674 3845043 default_node_network_controller.go:124] Enable node proxy healthz server on 0.0.0.0:10256
I0419 10:49:40.559687 3845043 default_node_network_controller.go:541] Starting the default node network controller
I0419 10:49:40.565625 3845043 default_node_network_controller.go:597] Node worker-51 ready for ovn initialization with subnet 10.131.0.0/23
I0419 10:49:40.565835 3845043 default_node_network_controller.go:448] Exporting management port annotation
I0419 10:49:40.566035 3845043 management-port-dpu.go:188] Lookup netdevice link and existing management port
I0419 10:49:40.566099 3845043 management-port-dpu.go:215] Setup netdevice management port: net041923104939
I0419 10:49:40.780275 3845043 ovs.go:200] Exec(1): /usr/sbin/sysctl -w net.ipv4.conf.ovn-k8s-mp0.forwarding=1
```

Reply:
Why do You think it shouldn't appear? If You just redeployed new image on already running node, then no wonder iface has name `net...`.

Reply:
After deploying the new image twice, you still expect the old `net...` name?

Reply:
Yes, because ovnkube doesn't know original VF name.

Reply:
ok, reboot should fix this?

Reply:
After reboot, the first time it works. Then, if I remove the daemonset and let it recover, it again sees the net... device name.